bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -0,0 +1,344 @@
+                            ((QuantityType<Energy>) command).toUnit(SmartHomeUnits.KILOWATT_HOUR).intValue() * 10);
+                }
+                break;
+            case ALLOW_CHARGING:
+                key = "alw";
+                if (command instanceof OnOffType) {
+                    value = command == OnOffType.ON ? "1" : "0";
+                }
+                break;
+            default:
+        }
+        if (key != null && value != null) {
+            sendData(key, value);
+        } else {
+            logger.warn("Could not update channel {} with key {} and value {}", channelUID.getId(), key, value);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(GoEChargerConfiguration.class);
+        allChannels = getThing().getChannels().stream().map(channel -> channel.getUID().getId())
+                .collect(Collectors.toList());
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        startAutomaticRefresh();
+        logger.debug("Finished initializing!");
+    }
+
+    private String getUrl(String type) {
+        String urlStr = type;
+        return urlStr.replace("%IP%", StringUtils.trimToEmpty(config.ip));
+    }
+
+    private void sendData(String key, String value) {
+        String urlStr = getUrl(GoEChargerBindingConstants.MQTT_URL).replace("%KEY%", key).replace("%VALUE%", value);
+        logger.debug("POST URL = {}", urlStr);
+
+        String result = null;
+        try {
+            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(HttpMethod.POST)
+                    .timeout(5, TimeUnit.SECONDS).send();
+
+            result = contentResponse.getContentAsString();
+            // TODO check if value really changed
+            logger.debug("Response: {}", result);
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "No response received on command");
+            return;
+        }
+
+        if (result != null) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            return;
+        }
+    }
+
+    /**
+     * Request new data from Go-E charger
+     *
+     * @return the Go-E charger object mapping the JSON response or null in case of
+     *         error
+     */
+    @Nullable
+    private GoEStatusResponseDTO getGoEData() {
+        GoEStatusResponseDTO result = null;
+
+        String urlStr = getUrl(GoEChargerBindingConstants.API_URL);
+        logger.debug("GET URL = {}", urlStr);
+
+        try {
+            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(HttpMethod.GET)
+                    .timeout(5, TimeUnit.SECONDS).send();
+
+            result = gson.fromJson(contentResponse.getContentAsString(), GoEStatusResponseDTO.class);
+            // logger.debug("Response: {}", contentResponse.getContentAsString());
```

Comment:
Please remove commented code or uncomment it and make it conditional `logger.isDebugEnabled()`

Reply:
Is the condition really necessary? It will log only with correct set log level, or not?